doc/fundamentals/history.md

```diff
@@ -0,0 +1,120 @@
+# History
+
+Pachyderm implements rich version-control and history semantics. This doc
+lays out the core concepts and architecture of Pachyderm's version-control
+and the various ways to use the system to access historical data.
+
+## Commits
+
+Commits are the core version-control primitive in Pachyderm, similar to
+git, commits represent an immutable snapshot of a filesystem and can be
+accessed with an ID. Commits have a parentage structure, with new commits
+inheriting content from their parents and then adding to it, you can think
+of it as a linked list, it's also often referred to as "a chain of
+commits." Commit IDs are useful if you want to have a static pointer to
+a snapshot of a filesystem. However, because they're static their use is
+limited, and you'll mostly deal with branches instead.
+
+## Branches
+
+Branches are tags that point to commits, again similar to git, branches
```

Comment:
Given [the "tags" feature request](https://github.com/pachyderm/pachyderm/issues/3598), could this be changed to just say 

> Branches are pointers to commits